### ISSUE TITLE
[Design Quality]: Update segmented tab styling

### ISSUE THREAD

**Issue by amandaye0h** [MetaMask/metamask-extension#26190]
### **Problem statement**

Visually, our tabs look unbalanced and compete for attention with other components. The updated styling strengthens the contrast between primary and secondary elements to help users focus stay focused on key tasks.

### **Acceptance criteria**

- Inactive tab text color is updated from `text-default` to `text-muted`
- Border styling for inactive tab =:` 2px solid color: var(--color-text-muted)`
- `4px` padding between tabs has been removed
- `font-weight = 500 (medium)` for active and inactive tabs

### **Testing**

A similar styling is applied in other flows:

- Send Flow (Your accounts/Contacts)
- Asset picker modal (Tokens/NFTs)
- Permissions page (Sites/Snaps)
- dApp connections (Connected accounts/Connected snaps)
- Import tokens (Search/Custom)

### **Final review**
- Review final implementation with @amandaye0h
- Notify @georgewrmarshall from Design Systems of changes

### **Key screen**
[Figma link](https://www.figma.com/design/bkJ9Ot0HGXiakMqZtQfs8G/Design-Quality?node-id=214-3)
<img width="440" alt="Screenshot 2024-07-29 at 7 38 10 AM" src="https://github.com/user-attachments/assets/e2a63981-231d-4bbd-9d0b-0bb4f46b9618">

### **Related screens**
[Figma link](https://www.figma.com/design/bkJ9Ot0HGXiakMqZtQfs8G/Design-Quality?node-id=221-38) 
<img width="1156" alt="Screenshot 2024-07-30 at 8 49 36 AM" src="https://github.com/user-attachments/assets/24674a8d-0ae9-4324-8fc4-2ae8a008e4d8">

### Steps to reproduce

**Portfolio View**
1. Open MetaMask

**Import tokens**
1. Open MetaMask
2. Scroll down
3. Click on `import tokens`

 **Send Flow**
1. Open MetaMask
2. Click on Send button
3. Click on the Asset selector (ETH) to access the modal

**Permissions**
(Associated with [Amon/Hen](https://www.figma.com/design/LDlI7yjKBNIY60MtQv9QS0/Amon-Hen%2FV1?node-id=9438-26704&t=WCJ8p039V9IkSecy-0))
1. Open MetaMask
2. Click on three dot menu in the top-right of the header
4. Click on Permissions

**dApp connection page**
1. Open MetaMask
2. Click on dApp icon in the header

### Detection stage

On the development branch

### Version

11.16.15

### Build type

None

### Browser

Brave

### Operating system

MacOS

**Comment by matteoscurati** [comment]
@amandaye0h I’m creating a PR to fix the tab styles as indicated here. I have two questions:
1. how should we handle cases where a tab list has only one tab? For now, I’ve created a style that applies text-default and border none (see image). Is this correct?
2. I believe some examples linked in Figma are outdated. For instance, the Snap section now has its own dedicated page and no longer lives under Permissions, correct?

<img width="358" alt="Image" src="https://github.com/user-attachments/assets/218daece-07a4-423f-a72c-1268cf6c0d13" />

**Comment by amandaye0h** [comment]
Thanks for working on this! 

1. In the case above, the tab should not exist. This was an implementation error.

<img width="750" alt="Image" src="https://github.com/user-attachments/assets/bbe65c5d-7271-4e2b-a71d-e54e15e20721" />

2. Yes that is correct. My bad for including the outdated designs!

**Comment by matteoscurati** [comment]
This is the PR with a proposed solution to standardize the tab style https://github.com/MetaMask/metamask-extension/pull/29652

**Comment by amandaye0h** [comment]
Yeah, I think it's an acceptable solution for now — best case scenario is that users don't see the tab at all. 

I'm ok with the white text if it's too complex to remove the tab altogether. The blue text + blue border is visually way too loud.
